### ISSUE TITLE
Point moves to hunk header after staging a hunk region

### ISSUE THREAD

**Issue by uu1101** [magit/magit#1855]
Point moves to the hunk header after staging a hunk region.  This makes it "laborious" when one wants to stage interspersed parts of a hunk.  It would make sense to restore point to beginning of the region or previous location (probably the former is easier) after the region has been applied.

This won't ever finish if for each closed issue I'm adding a new one :-).  In any case this is pretty much insignificant, just wanted to have it recorded.

This is on `next`.

Thanks,
uu1101.

**Comment by tarsius** [comment]
This is quite hard to do using the current approach, so that will have to wait until after the release.

**Comment by tarsius** [comment]
I have implemented something similar in #2227. That will probably make it into v2.3.

> It would make sense to restore point to beginning of the region or previous location (probably the former is easier) after the region has been applied.

Both are hard. The problem is that not only the added and removed lines disappear. Context lines may also appear or disappear and there might me more or fewer hunks than before. So instead I have opted to goto the beginning of the next or previous block of changed lines. That is easier to do because we know that those lines will appear in the new diff, exactly as they appeared in the old.
